### DISCUSSION TITLE
Macro for division

### DISCUSSION THREAD

**Discussion by johnpalermo** [c128lib/.github#5] [Team Posts]
Hello Team,

I was looking a the math library and did not see a macro for division of 2 integers.  @intoinside @c128lib/c128lib-dev-team I think it would be good to include it.  Any thoughts?

**Comment by intoinside** [comment]
Sorry for the late answer... I agree, could be a good starting point a math section. Do you plan to use some methods provided by the kernal?

**Comment by johnpalermo** [comment]
I could not find a routine for math division in the kernel.  I was thinking of adding a new macro to math-global.asm and math.asm in the common library.  I was not sure if you already had this in mind to do.  If not, I can do a pull request and add the macro.  I may need to use the division macro in the framework library.

**Comment by intoinside** [comment]
Ok proceed with add/pull request, there's no plan for common right now

Il gio 12 gen 2023, 17:52 John Palermo ***@***.***> ha
scritto:

**Comment by intoinside** [comment]
There are two macros planned (for cursor move and string output, so no math is involved).
These two are tagged for version 0.6.0. You can add you macro to math*.asm (and tag it with 0.6.0) so we can release this new version.

**Comment by johnpalermo** [comment]
Sounds good. I will do it this weekend. Very rough week at work.

------- Original Message -------

**Comment by johnpalermo** [comment]
Sorry for the late response.  I'm creating a fork now and will create a pull request once code is done.

**Comment by johnpalermo** [comment]
Hello.  I have created an issue to add 2 new macros for division - 16-bit/8-bit and 16-bit/16-bit.  You can assign it to me and label as enhancement.  Thanks.

**Comment by intoinside** [comment]
hi John, issue https://github.com/c128lib/common/issues/9 updated as your request, but i've a question for you. I need to understand if there is something missing on permission grant on repositories (i want developer to be autonomous, no need for me to grant something, no need for developer to wait for me) so:
* why you choose to fork, instead of creating a branch inside c128lib/common, and then creating pull request to master from it? 
VSCode-Github extension helps you to see issues and create a specific branch to work on, and also helps you to create a pull request.

**Comment by johnpalermo** [comment]
Ciao Raffaele,

No reason. It’s the way I know how to do it, but I think that your way sounds better. I will do it as you have suggested.

So, should I create issue in repository, then create branch, etc? I tried to update the issue I created, but I could only update the name and description. I couldn’t assign a label or a developer.

**Comment by intoinside** [comment]
I usually
create an issue
Start a new branch with issue id
Commit and  push on that branch
Pull request to master

While working on branch, i usually edit issue by adding branch I'm working
on so every commit/push is automatically linked.

When pull request is closed, i delete issue branch.

I've edited some organization access rules (i didn't fully understand) so
check if you can update your issue.

I think I'll create an "how to" with my best practice, adding also
extensions for VsCode which is quite helpful for me.

Il dom 22 gen 2023, 18:24 John Palermo ***@***.***> ha
scritto:

**Comment by johnpalermo** [comment]
Understood.  I will use the VSCode GitHub extension.  I agree that this is a better way.  I will create the branch from the issue using VSCode.  Then commit and push on that branch as you have mentioned.  Basically, I will follow your steps.  They are clear to me.  Yes, I think a "how to" is good.  People use different approaches and even use git cli, but it's good to be consistent with everyone on the team. 

Yes, the issue is now editable for me.  Thank you.

**Comment by intoinside** [comment]
Ok, i've created a first version of "best practice", it's availabel in c128lib project documentation. Direct link https://c128lib.github.io/bestpractice

I need to complete with pull request paragraph

**Comment by johnpalermo** [comment]
Pull request created and ready to merge into master.   
@intoinside I know that you allowed pull requests to be merged without approvals, but I made you the reviewer to test this GitHub feature and also to allow you to check the pull request before I merge it.
